video/out/opengl/context_drm_egl.c

```diff
@@ -497,9 +497,12 @@ static void drm_egl_uninit(struct ra_ctx *ctx)
 
         eglMakeCurrent(p->egl.display, EGL_NO_SURFACE, EGL_NO_SURFACE,
                        EGL_NO_CONTEXT);
-        eglDestroyContext(p->egl.display, p->egl.context);
-        eglDestroySurface(p->egl.display, p->egl.surface);
-        gbm_surface_destroy(p->gbm.surface);
+        if (p->egl.display) {
+            eglDestroyContext(p->egl.display, p->egl.context);
+            eglDestroySurface(p->egl.display, p->egl.surface);
+        }
+        if (p->gbm.surface)
+            gbm_surface_destroy(p->gbm.surface);
         eglTerminate(p->egl.display);
         gbm_device_destroy(p->gbm.device);
         p->egl.context = EGL_NO_CONTEXT;
```

Comment:
unrelated to this change but this looks curious:
<pre>
        p->egl.context = EGL_NO_CONTEXT;
        eglDestroyContext(p->egl.display, p->egl.context);
</pre>